### PR TITLE
fix(ha): add changes to ha feature helm components

### PR DESCRIPTION
**Pull request by Abhinandan-Purkait**
1. Fix helm template for ha-cluster.
2. Fix helm template for ha-node.
3. Enable ANA in io-engine template.
4. Add init-containers to HA node components.
5. Enable `hostNetwork` on HA node template for path detection to work.

Signed-off-by: Abhinandan Purkait <abhinandan.purkait@mayadata.io>

### CODE REVIEW

File: chart/templates/mayastor/agents/ha/ha-node-daemonset.yaml [hunk 1]
```diff
@@ -20,6 +20,10 @@ spec:
         app: agent-ha-node
         openebs.io/logging: "true"
     spec:
+      hostNetwork: true
```
Comment (mtzaurus) [inline chart/templates/mayastor/agents/ha/ha-node-daemonset.yaml:23]: This is required for using netlink sockets inside containers.

**Review by mtzaurus — APPROVED** [review]

File: chart/templates/mayastor/agents/ha/ha-node-daemonset.yaml [hunk 2]
```diff
@@ -57,7 +61,7 @@ spec:
         args:
         - "--node-name=$(MY_NODE_NAME)"
         - "--grpc-endpoint=$(MY_POD_IP):11600"
-        - "--cluster-agent=https://mayastor-agent-ha-cluster:11500"
+        - "--cluster-agent=https://mayastor-agent-core:11500"
```
Comment (avishnu) [inline chart/templates/mayastor/agents/ha/ha-node-daemonset.yaml:64]: Should this be {{ .RELEASE.NAME }}-agent-core instead of "mayastor-agent-core" ?

Comment (Abhinandan-Purkait) [reply chart/templates/mayastor/agents/ha/ha-node-daemonset.yaml:64]: Right. It should be. Will change it. Thank you.

**Review by abhilashshetty04 — APPROVED** [review]

File: chart/templates/mayastor/agents/ha/ha-node-daemonset.yaml (hunk 2 above)
Comment (Abhinandan-Purkait) [reply chart/templates/mayastor/agents/ha/ha-node-daemonset.yaml]: Done.

File: chart/templates/mayastor/agents/core/agent-core-deployment.yaml [hunk 3]
```diff
@@ -57,3 +57,34 @@ spec:
...
+              memory: {{ .Values.agents.ha.cluster.resources.limits.memory | quote }}
+            requests:
+              cpu: {{ .Values.agents.ha.cluster.resources.requests.cpu | quote }}
+              memory: {{ .Values.agents.ha.cluster.resources.requests.memory | quote }}
+          image: "{{ .Values.image.registry }}/{{ .Values.image.repo }}/mayastor-agent-ha-cluster:{{ .Values.image.tag }}"
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          args:
+            - "-g=0.0.0.0:11500"
+            - "--store=http://{{ .Release.Name }}-etcd:{{ .Values.etcd.service.port }}"
+            - "--core-grpc=https://mayastor-agent-core:50051"
```
Comment (tiagolobocastro) [inline chart/templates/mayastor/agents/core/agent-core-deployment.yaml:73]: "--core-grpc=https://{{ .Release.Name }}-agent-core:50051"

Comment (tiagolobocastro) [inline chart/templates/mayastor/agents/core/agent-core-deployment.yaml:73]: - "--jaeger={{ .Values.base.jaeger.agent.name }}:{{ .Values.base.jaeger.agent.port }}"{{ end }}

File: chart/templates/mayastor/agents/core/agent-core-service.yaml [hunk 4]
```diff
@@ -11,3 +11,5 @@ spec:
   ports:
     - name: grpc
       port: 50051
+    - name: ha-cluster
+      port: 11500
```
Comment (tiagolobocastro) [inline chart/templates/mayastor/agents/core/agent-core-service.yaml:15]: Should we use 50052?

File: chart/values.yaml [hunk 5]
```diff
@@ -31,6 +31,13 @@ base:
       - name: etcd-probe
         image: busybox:latest
         command: [ 'sh', '-c', 'trap "exit 1" TERM; until nc -vzw 5 {{ .Release.Name }}-etcd {{ .Values.etcd.service.port }}; do date; echo "Waiting for etcd..."; sleep 1; done;' ]
+  
+  initHaContainers:
```
Comment (tiagolobocastro) [inline chart/values.yaml:35]:
```suggestion
  initHaNodeContainers:
```

File: chart/templates/_helpers.tpl [hunk 6]
```diff
@@ -24,6 +24,17 @@ Usage:
     {{- end }}
 {{- end -}}
 
+{{/*
+Renders the HA NODE AGENT init container, if enabled
+Usage:
+{{ include "base_init_ha_containers" . }}
```
Comment (tiagolobocastro) [inline chart/templates/_helpers.tpl:30]: base_init_ha_node_containers

File: chart/templates/mayastor/agents/core/agent-core-deployment.yaml (hunk 3 above)
Comment (Abhinandan-Purkait) [reply chart/templates/mayastor/agents/core/agent-core-deployment.yaml]: Done.

Comment (Abhinandan-Purkait) [reply chart/templates/mayastor/agents/core/agent-core-deployment.yaml]: {{ if .Values.base.jaeger.enabled }}
            - "--jaeger={{ .Values.base.jaeger.agent.name }}:{{ .Values.base.jaeger.agent.port }}"{{ end }}``

File: chart/values.yaml (hunk 5 above)
Comment (Abhinandan-Purkait) [reply chart/values.yaml]: Done.

File: chart/templates/_helpers.tpl (hunk 6 above)
Comment (Abhinandan-Purkait) [reply chart/templates/_helpers.tpl]: Done.

File: chart/templates/mayastor/agents/core/agent-core-service.yaml (hunk 4 above)
Comment (Abhinandan-Purkait) [reply chart/templates/mayastor/agents/core/agent-core-service.yaml:15]: That would need changes from control-plane side as well. But yeah we can do it if needed

Comment (tiagolobocastro) [reply chart/templates/mayastor/agents/core/agent-core-service.yaml:15]: Why would it need control-plane changes?

**Review by tiagolobocastro — APPROVED** [review]

File: chart/templates/mayastor/agents/core/agent-core-service.yaml (hunk 4 above)
Comment (Abhinandan-Purkait) [reply chart/templates/mayastor/agents/core/agent-core-service.yaml:15]: Ohh it won't, I was thinking we need to change the constants in the control-plane but that is independent of this. Should I make the change then?

Comment (tiagolobocastro) [reply chart/templates/mayastor/agents/core/agent-core-service.yaml:15]: Up to you but I think so, would be good to keep the ports together, will also be easier to remember

Comment (Abhinandan-Purkait) [reply chart/templates/mayastor/agents/core/agent-core-service.yaml]: Done. Changed the node-agent's server port to be 50053 as well. Also added jaeger arg to node-agent

Comment (tiagolobocastro) [reply chart/templates/mayastor/agents/core/agent-core-service.yaml]: Nice and good catch on the node-agent jaeger!

**Comment by Abhinandan-Purkait** [comment]
bors merge

**Pull request merged**